hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py

```diff
@@ -0,0 +1,246 @@
+        return """
+            min(id) AS id,
+            employee_id,
+            department_id,
+            date,
+            sum(worked_hours) AS worked_hours,
+            max(theoretical_hours) AS theoretical_hours,
+            sum(difference) AS difference
+            """
+
+    def _select_sub1(self):
+        # Unique ID is assured (mostly, as we lose some precission) through
+        # this MD5 hash converted to integer. See
+        # https://stackoverflow.com/a/9812029 for details.
+        return """
+            (
+                ('x'||substr(MD5('HA' || ha.id::text), 1, 8))::bit(32)::int
+            ) AS id,
+            ha.employee_id AS employee_id,
+            hahe.department_id AS department_id,
+            ha.check_in::date AS date,
+            ha.worked_hours AS worked_hours,
+            ha.theoretical_hours AS theoretical_hours,
+            0.0 AS difference
+            """
+
+    def _from_sub1(self):
+        return """
+            hr_attendance ha
+            LEFT JOIN hr_employee AS hahe ON ha.employee_id = hahe.id
+            """
+
+    def _where_sub1(self):
+        return "True"
+
+    def _select_sub2(self):
+        # Same comment about ID uniqueness of sub1.
+        return """
+            (
+                ('x'||substr(MD5(
+                    'HE' || he.id::text || gs::text
+                ), 1, 8))::bit(32)::int
+            ) AS id,
+            he.id AS employee_id,
+            he.department_id AS department_id,
+            gs::date AS date,
+            0 AS worked_hours,
+            -1 AS theoretical_hours,
+            0.0 AS difference
+            """
+
+    def _from_sub2(self):
+        # We generate one record for each of the theoretical working days
+        # since the employee creation / working schedule beginning for not
+        # depending on the registered attendances.
+        return """
+                hr_employee he
+            INNER JOIN
+                resource_resource rr ON he.resource_id = rr.id
+            LEFT JOIN
+                resource_calendar_attendance rca
+                    ON rca.calendar_id = rr.calendar_id
+            CROSS JOIN
+                generate_series(
+                    greatest(
+                        COALESCE(he.theoretical_hours_start_date,
+                                 he.create_date::date),
+                        COALESCE(rca.date_from,
+                                 he.theoretical_hours_start_date,
+                                 he.create_date::date)
+                    )
+                    + (8 + rca.dayofweek::int -
+                        extract(dow from greatest(
+                            COALESCE(he.theoretical_hours_start_date,
+                                     he.create_date::date),
+                            COALESCE(rca.date_from,
+                                     he.theoretical_hours_start_date,
+                                     he.create_date::date)
```

Comment:
the tz then has already been manged [here](https://github.com/dzungtran89/hr-attendance/blob/16.0-mig-hr_attendance_report_theoretical_time/hr_attendance_report_theoretical_time/reports/hr_attendance_theoretical_time_report.py#L184)

Reply:
same as my previous comment, suppose that i live in HCM city, i create an attendance at 5am 2023/04/10.
I want the statistic report to count my attendance on  2023/04/10, but `he.create_date::date` = 2023/04/09, because it stores my attendance in UTC timezone ( create_date = 2023/04/09 05:00:00)

Reply:
I'm not sure why the author did not take the timezone into account, indeed it's a good point and we can fix that. But then it would be confusing with the native `hr.attendance.report`